static_frame/core/util.py

```diff
@@ -3186,5 +3187,91 @@ def iloc_to_insertion_iloc(key: int, size: int) -> int:
         raise IndexError(f'index {key} out of range for length {size} container.')
     return key % size
 
-
+class MemoryMeasurements:
+    @staticmethod
+    def _unsized_children(obj: tp.Any) -> tp.Iterable[tp.Any]:
+        '''
+        Generates the iterable children that have not been counted by a getsizeof call
+        on the parent object
+        '''
+        # Check if iterable or a string first for fewer isinstance calls on common types
+        if hasattr(obj, '__iter__') and not isinstance(obj, str):
+            if obj.__class__ is np.ndarray and obj.dtype.kind == DTYPE_OBJECT_KIND:
+                # _nested_sizable_elements only runs id(obj) in seen to check to skip elements
+                # If an element's id is the same as a previous element, and the element is different from
+                # the previous element, it would result in the element getting improperly skipped.
+                # By yielding from a list comprehension, each element in the numpy object array must be
+                # serialized to a PyObject first, ensuring that the ids are not reused.
+                # Example (Python 3.8.12/NumPy 1.19.5):
+                # >>> import numpy as np
+                # >>> a = np.array([np.array([None, None, i // 2]) for i in range(5)])
+                # >>> b = [id(a[0]), id(a[1]), id(a[2]), id(a[3]), id(a[4])]
+                # >>> c = [id(el) for el in a]
+                # >>> d = [el for el in a]
+                # >>> e = [id(el) for el in d]
+                # >>>
+                # >>> a
+                # array([[None, None, 0],
+                #        [None, None, 0],
+                #        [None, None, 1],
+                #        [None, None, 1],
+                #        [None, None, 2]], dtype=object)
+                # >>> b
+                # [139999659995264, 139999659995264, 139999659995264, 139999659995264, 139999659995264]
+                # >>> c
+                # [139999659995264, 139999658764368, 139999659995264, 139999658764368, 139999659995264]
+                # >>> d
+                # [array([None, None, 0], dtype=object), array([None, None, 0], dtype=object), array([None, None, 1], dtype=object), array([None, None, 1], dtype=object), array([None, None, 2], dtype=object)]
+                # >>> e
+                # [139999659995264, 139999658764368, 139999658765088, 139997430769344, 139997430769104]
+                yield from [el for el in obj]
```

Comment:
As discussed, I think this should do the same without the list... or you could use `iter(obj)`, but as you point out, we may get unexpected results if we have an object array that contains NumPy arrays. If this covers all cases better, fine to have this intermediary list, lets just make sure it is clear to the reader why.
```suggestion
                yield from obj
```

Reply:
I just realized that numpy is actually creating a multi-dimensional object array, even though I passed in a list of np.array objects to the constructor. The following will loop just on the objects referenced in the array.

Previously, numpy was creating new numpy arrays for each column as I looped through. This is likely why the ids were not consistent with the ids of the arrays I used in the constructor.

```
>>> import numpy as np
>>> a = np.array([np.array([None, None, i // 2]) for i in range(5)])
>>>
>>> a.shape
(5, 3)
>>>
>>> [id(a[loc]) for loc in np.ndindex(a.shape)]
[94341345798752, 94341345798752, 94341345978080, 94341345798752, 94341345798752, 94341345978080, 94341345798752, 94341345798752, 94341345978112, 94341345798752, 94341345798752, 94341345978112, 94341345798752, 94341345798752, 94341345978144]
```